src/UKHO.S100PermitService.Common/Services/PermitService.cs

```diff
@@ -11,45 +12,85 @@ public class PermitService : IPermitService
 
         private readonly ILogger<PermitService> _logger;
         private readonly IPermitReaderWriter _permitReaderWriter;
+        private readonly IHoldingsService _holdingsService;
 
         public PermitService(IPermitReaderWriter permitReaderWriter,
-                                ILogger<PermitService> logger)
+                                ILogger<PermitService> logger,
+                                IHoldingsService holdingsService)
         {
             _permitReaderWriter = permitReaderWriter ?? throw new ArgumentNullException(nameof(permitReaderWriter));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _holdingsService = holdingsService ?? throw new ArgumentNullException(nameof(holdingsService));
         }
 
-        public void CreatePermit(DateTimeOffset issueDate, string dataServerIdentifier, string dataServerName, string userPermit, decimal version, List<Products> products)
+        public async Task CreatePermitAsync(int licenceId, string correlationId)
+        {
+            _logger.LogInformation(EventIds.CreatePermitStart.ToEventId(), "CreatePermit started");
+
+            var holdingsServiceResponse = await _holdingsService.GetHoldingsAsync(licenceId, correlationId);
+
+            var productsList = GetProductsList();
+            const string Upn = "ABCDEFGHIJKLMNOPQRSTUVYXYZ";
+
+            CreatePermitXml(DateTimeOffset.Now, "AB", "ABC", Upn, 1.0m, productsList);
+
+            _logger.LogInformation(EventIds.CreatePermitEnd.ToEventId(), "CreatePermit completed");
+        }
+
+        private void CreatePermitXml(DateTimeOffset issueDate, string dataServerIdentifier, string dataServerName, string userPermit, decimal version, List<Products> products)
```

Comment:
Is version really implemented as a decimal in the spec?! I guess they aren't planning on having version 3.2.1! Could you check, and see if this could be made a string?

Reply:
verified with spec its string datatype. 
will make the required change.

Reply:
Resolved